pkg/sqlite/query.go

```diff
@@ -1117,3 +1117,71 @@ func (s *SQLQuerier) GetPropertiesForBundle(ctx context.Context, name, version,
 
 	return
 }
+
+func (s *SQLQuerier) GetTailForBundleInChannel(ctx context.Context, csvName string) (bundleNames []string, err error) {
+	getDepthQuery := `SELECT DISTINCT channel_name, package_name, depth FROM channel_entry WHERE operatorbundle_name=?`
+
+	getTailQuery := `SELECT DISTINCT operatorbundle_name FROM channel_entry 
+				WHERE channel_name=?
+				AND package_name=?
+				AND depth > ?`
+
+	rows, err := s.db.QueryContext(ctx, getDepthQuery, csvName)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	tail := make(map[string]struct{})
+
+	for rows.Next() {
+		var channel sql.NullString
+		var pkg sql.NullString
+		var depth sql.NullInt64
+		if err := rows.Scan(&channel, &pkg, &depth); err != nil {
+			return nil, err
+		}
+
+		tailRows, err := s.db.QueryContext(ctx, getTailQuery, channel, pkg, depth)
+		if err != nil {
+			return nil, err
+		}
+		defer tailRows.Close()
```

Comment:
IIRC, it's racy to call `Close()` on a row object while another one (in this case, `rows`) is still not `Close`d.